src/traces/barpolar/hover.js

```diff
@@ -0,0 +1,78 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var barHover = require('../bar/hover');
+var makeHoverPointText = require('../scatterpolar/hover').makeHoverPointText;
+var Color = require('../../components/color')
+var Fx = require('../../components/fx');
+var Lib = require('../../lib');
+var isPtInsidePolygon = require('../../plots/polar/helpers').isPtInsidePolygon;
+
+module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
+    var cd = pointData.cd;
+    var trace = cd[0].trace;
+    var t = cd[0].t;
+
+    var subplot = pointData.subplot;
+    var radialAxis = subplot.radialAxis;
+    var angularAxis = subplot.angularAxis;
+    var xa = subplot.xaxis;
+    var ya = subplot.yaxis;
+    var inboxFn = subplot.vangles ? isPtInsidePolygon : Lib.isPtInsideSector;
+
+    if(radialAxis.range[0] > radialAxis.range[1]) {
+        xval = -xval;
+        yval = -yval;
+    }
```

Comment:
This confuses me... I see you have a test for it so I guess I can't complain too much, but wouldn't it leave `rVal` unchanged while flipping `thetaVal` 180º?

Reply:
> but wouldn't it leave rVal unchanged while flipping thetaVal 180º?

Good call. That's much easier to read -> https://github.com/plotly/plotly.js/pull/2954/commits/971c6f6386c2c73a7dbd993d5953be842eb784e5

Reply:
OK, I still don't get why reversed *radial* range needs a change in *angle* (please don't bother explaining it 🙈 ) but it's clear anyway - the commit that I commented on didn't have the comment above this, that helps a lot.